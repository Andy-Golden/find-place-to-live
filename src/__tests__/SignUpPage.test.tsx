import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { fireEvent, render, screen } from "@testing-library/react";

import { SignUpPage } from "@pages";
import { ToastProvider } from "@providers";
import { store } from "@store";
import renderer from "react-test-renderer";

import "@testing-library/jest-dom";

describe("SelectMenu component", () => {
  it("Snap shot", () => {
    const tree = renderer
      .create(
        <ToastProvider>
          <Provider store={store}>
            <BrowserRouter>
              <SignUpPage />
            </BrowserRouter>
          </Provider>
        </ToastProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Form should be submitted", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <SignUpPage />
        </BrowserRouter>
      </Provider>,
    );

    const handleOnSubmitMock = jest.fn();

    screen.getByRole("form").onsubmit = handleOnSubmitMock;

    fireEvent.change(screen.getByPlaceholderText("signUpPage.enterFullname"), {
      target: { value: "validUsername" },
    });
    fireEvent.change(screen.getByPlaceholderText("enterEmail"), {
      target: { value: "validemail@example.com" },
    });
    fireEvent.change(screen.getByPlaceholderText("enterPassword"), {
      target: { value: "Hello@1" },
    });
    fireEvent.change(
      screen.getByPlaceholderText("signUpPage.enterConfirmation"),
      {
        target: { value: "Hello@1" },
      },
    );

    fireEvent.click(screen.getByRole("button", { name: "signUp" }));

    expect(handleOnSubmitMock).toHaveBeenCalled();
  });
});
