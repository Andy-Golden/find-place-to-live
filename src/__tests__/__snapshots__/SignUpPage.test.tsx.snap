// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu component Snap shot 1`] = `
[
  <div
    className="auth-container"
  >
    <form
      autoComplete="off"
      className="register-form"
      name="sign-up-form"
      onSubmit={[Function]}
    >
      <div
        className="lang-form-layout"
      >
        <div
          className="select-menu "
          style={
            {
              "background": undefined,
              "color": undefined,
            }
          }
        >
          <select
            className="select undefined"
            onChange={[Function]}
            value="en"
          >
            <option
              value="en"
            >
              en
            </option>
            <option
              value="vi"
            >
              vi
            </option>
          </select>
        </div>
      </div>
      <div
        className="register-form__heading"
      >
        <h1
          className="title"
        >
          signUp
        </h1>
        <p
          className="subtitle"
        >
          signUpPage.subTitle
        </p>
      </div>
      <div
        className="register-form__input-boxes"
      >
        <div
          className="single-input-fields"
        >
          <label>
            signUpPage.fullname
          </label>
          <input
            className="input "
            name="fullname"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="signUpPage.enterFullname"
            style={
              {
                "background": undefined,
                "color": undefined,
              }
            }
            type="text"
          />
        </div>
        <div
          className="single-input-fields"
        >
          <label>
            email
          </label>
          <input
            className="input "
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="enterEmail"
            style={
              {
                "background": undefined,
                "color": undefined,
              }
            }
            type="text"
          />
        </div>
        <div
          className="single-input-fields"
        >
          <label>
            password
          </label>
          <input
            className="input "
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="enterPassword"
            style={
              {
                "background": undefined,
                "color": undefined,
              }
            }
            type="password"
          />
        </div>
        <div
          className="single-input-fields"
        >
          <label>
            signUpPage.confirmation
          </label>
          <input
            className="input "
            name="confirmation"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="signUpPage.enterConfirmation"
            style={
              {
                "background": undefined,
                "color": undefined,
              }
            }
            type="password"
          />
        </div>
      </div>
      <div
        className="register-form__footer"
      >
        <p>
          signUpPage.alreadyHaveAccount
           
          <a
            href="/login"
            onClick={[Function]}
          >
            signIn
          </a>
           
          signUpPage.here
        </p>
        <button
          className="button "
          style={
            {
              "background": undefined,
              "color": undefined,
            }
          }
          type="submit"
        >
          signUp
        </button>
      </div>
    </form>
  </div>,
  " ",
  <div
    className="toasts"
  />,
]
`;
